src/main/java/org/takes/http/BkBasic.java

```diff
@@ -73,16 +86,28 @@ public BkBasic(final Take tks) {
     @Override
     public void accept(final Socket socket) throws IOException {
         final InputStream input = socket.getInputStream();
+        final RqLive req = new RqLive(input);
+        boolean keep = false;
+        final Iterator<String> values = new RqHeaders.Base(req)
+            .header(CONNECTION).iterator();
+        if (values.hasNext()) {
+            do {
+                keep = KEEP_ALIVE.equals(values.next());
```

Comment:
@lautarobock if keepalive is present but then another value is present in the `Connection` header, `keep` will be false. We should have an `if` here